oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java

```diff
@@ -66,97 +71,200 @@ public IndexPrinter(IndexInfoService indexInfoService, AsyncIndexInfoService asy
-            pw.printf("    Last indexed up to      : %s%n", formatTime(info.getIndexedUpToTime()));
+            keyValue("    Last indexed up to       ", formatTime(info.getIndexedUpToTime()), pw, json, format);
         }
 
         if (info.getLastUpdatedTime() > 0){
-            pw.printf("    Last updated time       : %s%n", formatTime(info.getLastUpdatedTime()));
+            keyValue("     Last updated time       ", formatTime(info.getLastUpdatedTime()), pw, json, format);
+        }
+
+        if (info.getCreationTimestamp() > 0){
+            keyValue("     Creation time           ", formatTime(info.getCreationTimestamp()), pw, json, format);
+        }
+
+        if (info.getReindexCompletionTimestamp() > 0){
+            keyValue("     Reindex completion time ", formatTime(info.getReindexCompletionTimestamp()), pw, json, format);
         }
 
         if (info.getSizeInBytes() >= 0){
-            pw.printf("    Size                    : %s%n", IOUtils.humanReadableByteCount(info.getSizeInBytes()));
+            keyValue("    Size                     ", IOUtils.humanReadableByteCount(info.getSizeInBytes()), pw, json, format);
+            keyValue("    Size (in Bytes)          ", info.getSizeInBytes(), pw, json, format);
+        }
+
+        if (info.getSuggestSizeInBytes() >= 0){
+            keyValue("    Suggest size             ", IOUtils.humanReadableByteCount(info.getSuggestSizeInBytes()), pw, json, format);
+            keyValue("    Suggest size (in Bytes)  ", info.getSuggestSizeInBytes(), pw, json, format);
         }
 
         if (info.getEstimatedEntryCount() >= 0){
-            pw.printf("    Estimated entry count   : %d%n", info.getEstimatedEntryCount());
+            keyValue("    Estimated entry count    ", info.getEstimatedEntryCount(), pw, json, format);
+        }
+
+        if ("lucene".equals(info.getType())) {
+            // Only valid for lucene type indexes, for others it will simply show false.
+            keyValue("    Has hidden oak mount     ", info.hasHiddenOakLibsMount(), pw, json, format);
+            keyValue("    Has property index       ", info.hasPropertyIndexNode(), pw, json, format);
         }
 
         if (info.hasIndexDefinitionChangedWithoutReindexing()) {
-            pw.println("    Index definition changed without reindexing");
             String diff = info.getIndexDefinitionDiff();
             if (diff != null) {
-                pw.println("    "+diff);
+                keyValue("    Index definition changed without reindexing ", diff, pw, json, format);
+                printWithNewLine(pw, "", format);
             }
         }
-        pw.println();
+        endJsonObject(json);
     }
 
     private static String formatTime(long time){
         Calendar cal = Calendar.getInstance();
         cal.setTimeInMillis(time);
-        return ISO8601.format(cal);
+        Date date = cal.getTime();
+        SimpleDateFormat outputFmt = new SimpleDateFormat("yyyy-mm-dd'T'HH:mm:ss.s z");
```

Comment:
mm is for minutes, but you need month, I think that would be MM. "ss.s" is incorrect, it shows seconds twice. I don't think we need milliseconds or fractional seconds, but if you want them then this needs to be fixed.

For human readable, what about

    SimpleDateFormat outputFmt = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss 'UTC'");

For ISO 8601, I think this would be correct:

    SimpleDateFormat outputFmt = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");

Reply:
Changed the format to look like this now 
`2021-09-21T07:16:53 UTC`

Reply:
`2021-09-21T07:16:53Z` Set format to 
`SimpleDateFormat outputFmt = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");`